library/std/src/sys/uefi/common.rs

```diff
@@ -0,0 +1,269 @@
+//! Contains most of the shared UEFI specific stuff. Some of this might be moved to `std::os::uefi`
+//! if needed but no point in adding extra public API when there is not Std support for UEFI in the
+//! first place
+
+use r_efi::efi::Guid;
+
+use crate::io::{self, const_io_error};
+use crate::mem::MaybeUninit;
+use crate::os::uefi;
+use crate::ptr::NonNull;
+
+// Locate handles with a particular protocol GUID
+/// Implemented using `EFI_BOOT_SERVICES.LocateHandles()`
+pub(crate) fn locate_handles(mut guid: Guid) -> io::Result<Vec<NonNull<crate::ffi::c_void>>> {
```

Comment:
As far as I know, UEFI does not specify what these Handles are supposed to be. It is perfectly fine for a UEFI implementation to store handles as integers internally.

Reply:
Fair enough!

Reply:
I guess we do still have the option of using
```rust
extern {
    type UefiHandle;
}
```
if memory serves.

Reply:
Extern types seem to be quite a good option. I do have a few reservations though:
1. The UEFI spec defines `typedef VOID *EFI_HANDLE;`. While one can argue that this is done due to just following C convention of Opaque types, this does mean most people coming from outside Rust will be expecting it to be just a Void pointer. Again, maybe this can be solved by just good documentation.
2. Extern types are not stable yet. Due to this most crates outside std might not want to use them yet. Eg: r-efi.

Thus I don't think we should convert to using extern types as long as we use r-efi. Maybe it can be done once extern types stabilize in the future.